Match filter

yuriyoleynikov opened the pull request.

yuriyoleynikov committed:
This changes:
-Add FilterAsynbc function in HomeController
-Add fild in ContainerForMatchModel class
-Add Filter block in Index view
-Add styles in site.css

yuriyoleynikov committed:
-Add GetMatches3 with filter

yuriyoleynikov committed:
Change GetMatches to GetMathes3

yuriyoleynikov committed:
Add filter functions

yuriyoleynikov committed:
Filter for Players modifications

yuriyoleynikov committed:
Add state filter
Add ContainerForMatchModel class filds

yuriyoleynikov committed:
Fix state filter

yuriyoleynikov committed:
Fix Index view

Igorbek commented on src/YoTennis/Controllers/HomeController.cs:
```diff
@@ -85,22 +85,25 @@ public static string Separate(MatchModel state)
             {
                 var match = await _matchListService.GetMatchService(UserId, id);
-                var state = await match.GetStateAsync();
+                var cur_state = await match.GetStateAsync();
```
do not use `snake_case`, use `camelCase` for local variables

Igorbek commented on src/YoTennis/Controllers/HomeController.cs:
```diff
@@ -109,12 +112,39 @@ public static string Separate(MatchModel state)
 
+        public async Task<List<string>> FilterAsync()
+        {
+            List<string> FPlayers = new List<string>();
```
`camelCase`

Igorbek replied:
you can use just `var`

yuriyoleynikov committed:
Merge branch 'master' into match-filter

yuriyoleynikov committed:
-Use class EnumerableExtensions in Index view

yuriyoleynikov committed:
-Add tests
-Add MatchInfo class
-Add Conversion class
-Method Separate to new Class ForScoreExtensions

yuriyoleynikov committed:
Fix names class and methods

Igorbek reviewed (changes requested):
Please address my previous comments

yuriyoleynikov committed:
Add new table in database (MatchInfos)

yuriyoleynikov committed:
Refactor ToMatchInfo method

yuriyoleynikov committed:
-Add RebuildMatchInfosAsync Metod
-Add GetPlayersAsync Metod
-Change name to GetMatchesWithFilter

yuriyoleynikov committed:
Use new methods in HomeController

yuriyoleynikov committed:
Fix reduilding MatchInfo
